sourcegraph/sourcegraph
# Cody Recipe Widget for Code Snippet Responses

Closes #55189. Paired on with @st0nebraker and @thenamankumar.

- Adds cody recipe widget to chat code snippets
- Opens `CodyRecipesWidget.tsx` to be used outside of CodeMirror; Search Results view should be able to use this widget by sending a `CodyChatStore` instance & `editor` (separate editor instance should be created)

## Depedency
- Merge /Cody PR [#445](https://github.com/sourcegraph/cody/pull/445) first

## Demo
https://github.com/sourcegraph/sourcegraph/assets/59381432/4a5f8378-ca9d-48fc-a242-005a87ee6b00

https://github.com/sourcegraph/sourcegraph/assets/59381432/94d6f4f2-2e56-4d17-8e38-a007b60a2d09

## Test plan
- Open up Cody sidebar chat from the Blob view, select a code snippet and observe the recipes coming into view
- Navigate to `/cody` page and repeat above steps
